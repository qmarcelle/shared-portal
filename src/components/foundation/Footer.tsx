import Image from 'next/image';
import Link from 'next/link';
import React from 'react';

// Define TypeScript interface for any props if needed in the future
interface FooterProps {}

const Footer: React.FC<FooterProps> = () => {
  const currentYear = new Date().getFullYear(); // Gets the current year
  return (
    <footer className="bg-neutral p-5 text-base-100">
      <div className="container mx-auto grid grid-cols-1 md:grid-cols-3 lg:grid-cols-4 gap-5">
        {/* Popular Links */}
        <div>
          <h5 className="text-lg font-bold mb-2">Popular Links</h5>
          <ul className="list-none">
            <li>
              <Link href="/id-card" className="link link-hover">
                Get an ID Card
              </Link>
            </li>
            <li>
              <Link href="/find-care" className="link link-hover">
                Find Care & Costs
              </Link>
            </li>
            <li>
              <Link href="/claims" className="link link-hover">
                View Claims
              </Link>
            </li>
            <li>
              <Link href="/profile" className="link link-hover">
                Profile Settings
              </Link>
            </li>
          </ul>
        </div>

        {/* Support */}
        <div>
          <h5 className="text-lg font-bold mb-2">Support</h5>
          <ul className="list-none">
            <li>
              <Link href="/contact" className="link link-hover">
                Get Help & Contact Us
              </Link>
            </li>
            <li>
              <Link href="/feedback" className="link link-hover">
                Share Your Opinions
              </Link>
            </li>
            <li>
              <Link href="/screen-share" className="link link-hover">
                Share Your Screen
              </Link>
            </li>
          </ul>
        </div>

        {/* Important Information */}
        <div>
          <h5 className="text-lg font-bold mb-2">Important Information</h5>
          <ul className="list-none">
            <li>
              <Link href="/nondiscrimination" className="link link-hover">
                Nondiscrimination
              </Link>
            </li>
            <li>
              <Link href="/fraud" className="link link-hover">
                Fight Fraud
              </Link>
            </li>
          </ul>
        </div>

        {/* Download the App */}
        <div>
          <h5 className="text-lg font-bold mb-2">Download the App</h5>
          <p>Download the app in the Apple Store or Google Play.</p>
          <div className="flex space-x-2">
            <a
              href="https://apple.com"
              className="link link-hover"
              aria-label="Download on the Apple Store"
            >
              <Image src="/path-to-apple-store-logo.png" alt="Apple Store" />
            </a>
            <a
              href="https://play.google.com"
              className="link link-hover"
              aria-label="Download on Google Play"
            >
              <Image src="/path-to-google-play-logo.png" alt="Google Play" />
            </a>
          </div>
        </div>
      </div>

      {/* Lower Footer */}
      <div className="border-t border-neutral-focus mt-5 pt-3 text-sm">
        <div className="container mx-auto grid grid-cols-1 md:grid-cols-2">
          <p>Â©1998-{currentYear} BlueCross BlueShield of Tennessee, Inc.</p>
          <div className="flex justify-end space-x-2">
            <Link href="/sitemap" className="link link-hover">
              Sitemap
            </Link>
            <Link href="/privacy" className="link link-hover">
              Privacy & Security
            </Link>
            <Link href="/legal" className="link link-hover">
              Legal
            </Link>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
