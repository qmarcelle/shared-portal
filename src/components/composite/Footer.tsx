import { useGroup } from '@/app/providers/GroupProvider';
import { IComponent } from '../IComponent';
import { Column } from '../foundation/Column';
import { Row } from '../foundation/Row';
import { TextBox } from '../foundation/TextBox';

export interface FooterProps extends IComponent {}

export function Footer({ className }: FooterProps) {
  const { group } = useGroup();
  const currentYear = new Date().getFullYear();

  return (
    <footer className={`w-full bg-gray-100 py-8 ${className ?? ''}`}>
      <Column className="app-content">
        <Row className="justify-between items-center">
          <TextBox text={`Â© ${currentYear} All rights reserved`} />
          <Row className="space-x-4">
            <TextBox
              className="hover:underline cursor-pointer"
              text="Privacy Policy"
            />
            <TextBox
              className="hover:underline cursor-pointer"
              text="Terms of Use"
            />
            <TextBox
              className="hover:underline cursor-pointer"
              text="Contact Us"
            />
          </Row>
        </Row>
      </Column>
    </footer>
  );
}
