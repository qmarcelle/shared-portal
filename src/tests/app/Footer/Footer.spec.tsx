import Footer from '@/components/foundation/Footer';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
const renderUI = () => {
  return render(<Footer />);
};

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      prefetch: () => null,
      replace: () => null,
    };
  },
}));

describe('Footer Component', () => {
  // Test: Ensure Popular Links section renders correctly
  const component = renderUI();
  it('should render the UI correctly', async () => {
    expect(screen.getByText('Popular Links')).toBeInTheDocument();
    expect(screen.getByText('Get an ID Card')).toBeInTheDocument();
    expect(screen.getByText('Find Care & Costs')).toBeInTheDocument();
    expect(screen.getByText('View Claims')).toBeInTheDocument();
    expect(screen.getByText('Profile Settings')).toBeInTheDocument();
    expect(screen.getByText('Support')).toBeInTheDocument();
    expect(screen.getByText('Get Help & Contact Us')).toBeInTheDocument();
    expect(screen.getByText('Share Website Feedback')).toBeInTheDocument();
    expect(screen.getByText('Share Your Screen')).toBeInTheDocument();
    expect(screen.getByText('Important Information')).toBeInTheDocument();
    expect(screen.getByText('Nondiscrimination')).toBeInTheDocument();
    expect(screen.getByText('Member Rights')).toBeInTheDocument();
    expect(screen.getByText('Fight Fraud')).toBeInTheDocument();

    expect(screen.getByText('Download the App')).toBeInTheDocument();
    expect(
      screen.getByText(
        /Download the BCBSTN app in the Apple Store or Google Play/i,
      ),
    ).toBeInTheDocument();

    const socialMediaIcons = screen.getAllByRole('img');
    expect(socialMediaIcons.length).toBeGreaterThanOrEqual(6); // Check if at least 6 icons are rendered
    expect(screen.getByAltText('Facebook Icon')).toBeInTheDocument();

    expect(screen.getByText('Sitemap')).toBeInTheDocument();
    expect(screen.getByText('Privacy & Security')).toBeInTheDocument();
    expect(screen.getByText('Legal')).toBeInTheDocument();
    expect(screen.getByText('Deutsch')).toBeInTheDocument();

    const user = userEvent.setup();
    const getIdCard = screen.getByText('Get an ID Card');
    expect(getIdCard).toHaveAttribute('href', '/memberIDCard');
    await user.click(getIdCard);

    const getFindCare = screen.getByText('Find Care & Costs');
    expect(getFindCare).toHaveAttribute('href', '/findcare');
    await user.click(getFindCare);

    const getClaims = screen.getByText('View Claims');
    expect(getClaims).toHaveAttribute('href', '/claims');
    await user.click(getClaims);

    const getSupport = screen.getByText('Get Help & Contact Us');
    expect(getSupport).toHaveAttribute('href', '/support');
    await user.click(getSupport);

    const getProfileSettings = screen.getByText('Profile Settings');
    expect(getProfileSettings).toHaveAttribute('href', '/profileSettings');
    await user.click(getProfileSettings);

    expect(component.baseElement).toMatchSnapshot();
  });
});
