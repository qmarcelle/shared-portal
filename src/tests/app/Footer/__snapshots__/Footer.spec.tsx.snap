// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render the UI correctly 1`] = `
<body>
  <div>
    <footer>
      <section
        class="brand-gradient"
        role="contentinfo"
      >
        <div
          class="flex flex-col md:text-left text-center app-content container mx-auto grid grid-cols-1 md:grid-cols-[1fr_1fr_1fr_29%] lg:grid-cols-[1fr_1fr_1fr_40%] gap-4 md:gap-x-8 text-sm text-white min-h-[216px]"
        >
          <div
            class="flex flex-col"
          >
            <h3
              class="body-bold mb-4"
            >
              Popular Links
            </h3>
            <ul
              class="space-y-4 font-thin body-2"
            >
              <li>
                <a
                  href="/member/idcard"
                >
                  Get an ID Card
                </a>
              </li>
              <li>
                <a
                  href="/member/findcare"
                >
                  Find Care & Costs
                </a>
              </li>
              <li>
                <a
                  href="/member/myplan/claims"
                >
                  View Claims
                </a>
              </li>
              <li>
                <a
                  href="/member/profile"
                >
                  Profile Settings
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col"
          >
            <h3
              class="body-bold mb-4"
            >
              Support
            </h3>
            <ul
              class="space-y-4 body-2 font-thin"
            >
              <li>
                <a
                  href="/member/support"
                >
                  Get Help & Contact Us
                </a>
              </li>
              <li>
                <a
                  href="http://wn008422.bcbst.com:81/portal/portal_refresh1/index.html#987736460"
                >
                  Share Website Feedback
                </a>
              </li>
              <li>
                <a
                  href="http://wn008422.bcbst.com:81/portal/portal_feature_cobrowse/index.html#988880562"
                >
                  Share Your Screen
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col"
          >
            <h3
              class="body-bold mb-4"
            >
              Important Information
            </h3>
            <ul
              class="space-y-4 body-2 font-thin"
            >
              <li>
                <a
                  href="https://www.bcbst.com/about/our-company/corporate-governance/legal/nondiscrimination-notice"
                >
                  Nondiscrimination
                </a>
              </li>
              <li>
                <a
                  href="https://www.bcbst.com/use-insurance/member-rights"
                >
                  Member Rights
                </a>
              </li>
              <li>
                <a
                  href="https://www.bcbst.com/fraud"
                >
                  Fight Fraud
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex flex-col md:ml-16 p-7 md:p-0 text-left"
          >
            <div
              class="flex flex-row md:block lg:flex"
            >
              <img
                alt=""
                class=""
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <div
                class="flex flex-col ml-4 md:ml-0 lg:ml-4 mt-1"
              >
                <h3
                  class="body-bold mb-2"
                >
                  Download the App
                </h3>
                <p
                  class="body-2"
                >
                  Download the BCBSTN app in the Apple Store or Google Play.
                </p>
              </div>
            </div>
            <div
              class="flex flex-row flex mt-4 grid grid-cols-6 md:grid-cols-3 md:grid-rows-2 lg:grid-cols-6 md:mb-2 mb-0 lg:mb-0 gap-x-2 gap-y-4"
            >
              <a
                href="https://www.facebook.com/bcbst"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Facebook Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
              <a
                href="https://www.linkedin.com/company/bcbst"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Linkedin Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
              <a
                href="https://www.instagram.com/bcbst/?hl=en"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Instagram Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
              <a
                href="https://www.youtube.com/@bcbstennessee/shorts"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Youtube Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
              <a
                href="https://www.pinterest.com/bcbst/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Pinterest Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
              <a
                href="https://twitter.com/bcbst"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="X Icon"
                  data-nimg="1"
                  decoding="async"
                  height="35"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="35"
                />
              </a>
            </div>
          </div>
        </div>
      </section>
      <section
        class="bg-neutral text-white text-xs"
      >
        <div
          class="flex flex-col app-content container mx-auto text-center space-y-8 mb-[72px]"
        >
          <div
            class="flex flex-row flex flex-wrap justify-center gap-2 body-2"
          >
            <a
              href=""
            >
              Español
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              العربية
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              繁體中文
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Tiếng Việt
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              한국어
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Français
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              ພາສາລາວ
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Deutsch
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              ગુજરાતી
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              日本語
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Tagalog
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              हिंदी
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Русский
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              فارسی
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Kreyòl Ayisyen
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Polski
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Português
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Italiano
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Diné Bizaad
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Deitsch
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Gagana Samoa
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Kapasal Falawasch
            </a>
            <span>
              |
            </span>
            <a
              href=""
            >
              Chamoru
            </a>
          </div>
          <div
            class="flex flex-row mt-8 border-t border-text-white flex-col-reverse lg:flex-row pt-4 flex lg:justify-between space-y-2 md:space-y-0 lg:pb-4 body-2"
          >
            <div
              class="flex flex-col lg:w-[67%] lg:text-left mt-4"
            >
              ©1998-
              2025
               BlueCross BlueShield of Tennessee, Inc., an Independent Licensee of the Blue Cross Blue Shield Association. BlueCross BlueShield of Tennessee is a Qualified Health Plan issuer in the Health Insurance Marketplace. 1 Cameron Hill Circle, Chattanooga TN 37402-0001
            </div>
            <div
              class="flex flex-row space-x-4 body-2 !mt-4 lg:text-left justify-center"
            >
              <a
                href=""
              >
                Sitemap
              </a>
              <span>
                |
              </span>
              <a
                href="https://www.bcbst.com/about/our-company/corporate-governance/privacy-security"
              >
                Privacy & Security
              </a>
              <span>
                |
              </span>
              <a
                href="https://www.bcbst.com/about/our-company/corporate-governance/legal"
              >
                Legal
              </a>
            </div>
          </div>
        </div>
      </section>
    </footer>
  </div>
</body>
`;
